terraform/services/ecs_service/alb.tf

```diff
@@ -12,7 +12,8 @@ resource "aws_alb_target_group" "ecs_service" {
   }
 }
 
-resource "aws_alb_listener_rule" "rule" {
+resource "aws_alb_listener_rule" "path_rule" {
+  count        = "${1 - var.use_host_condition}"
```

Comment:
Nice. Didn't know these existed. Apparently they came in v0.8: https://www.hashicorp.com/blog/terraform-0-8/#conditional